gnocchi/rest/aggregates/api.py

```diff
@@ -255,24 +268,37 @@ def groupper(r):
                 api.enforce("get metric", metric)
 
             metrics_by_ids = dict((six.text_type(m.id), m) for m in metrics)
-            metrics_and_aggregations = [(metrics_by_ids[m], a)
-                                        for (m, a) in references]
-            return get_measures_or_abort(
-                metrics_and_aggregations, body["operations"],
-                start, stop, granularity, needed_overlap, fill,
-                ref_identifier="id")
+            references = [processor.MetricReference(metrics_by_ids[m], a)
+                          for (m, a) in references]
+
+            response = {
+                "measures": get_measures_or_abort(
+                    references, body["operations"],
+                    start, stop, granularity, needed_overlap, fill)
+            }
+            if details:
+                response["references"] = references
+            return response
 
     def _get_measures_by_name(self, resources, metric_names, operations,
-                              start, stop, granularity, needed_overlap, fill):
+                              start, stop, granularity, needed_overlap, fill,
+                              details):
+
+        references = list(map(lambda x: processor.MetricReference(*x),
+                              filter(lambda x: x[0] is not None,
+                                     ((r.get_metric(metric_name), agg, r)
+                                      for (metric_name, agg) in metric_names
+                                      for r in resources))))
```

Comment:
do we need this filter or can we just do:

```
processor.MetricReference(r.get_metric(metric_name), agg, r)
for (metric_name, agg) in metric_names
for r in resources if r.get_metric(metric_name) is not None
```

Reply:
That's work too and this is more readable.